metaseq/benchmark/generator.py

```diff
@@ -0,0 +1,136 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates. All Rights Reserved.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import logging
+import random
+import socket
+import sys
+from typing import Tuple
+
+import numpy as np
+import torch
+from torch.profiler.profiler import (
+    ProfilerActivity,
+    schedule,
+    tensorboard_trace_handler,
+)
+from tqdm import tqdm
+
+from metaseq import options
+from metaseq.dataclass.configs import MetaseqConfig
+from metaseq.dataclass.utils import convert_namespace_to_omegaconf
+from metaseq.distributed import utils as distributed_utils
+from metaseq.hub_utils import GeneratorInterface
+
+logging.basicConfig(format="%(asctime)s | %(name)s | %(message)s", level=logging.INFO)
+logger: logging.Logger = logging.getLogger("metaseq.benchmark.generator")
+
+WARMUP_STEPS = 5
+NUM_REPEATS = 5
+generator = None
+
+
+def main(cfg: MetaseqConfig) -> None:
+    global generator
+    torch.manual_seed(random.randint(1, 20000))
+
+    logger.info("Instantiating a generator interface and loading a model")
+    generator = GeneratorInterface(cfg)
+    generator.load_model()
+
+    batch_size = cfg.dataset.batch_size
+    input_length = cfg.task.max_source_positions
+    output_length = cfg.task.max_target_positions
+
+    logger.info("Running warm-up steps before benchmarking")
+    for _ in range(WARMUP_STEPS):
+        inputs = np.random.randint(100, 50000, size=(batch_size, input_length))
```

Comment:
Where did 100-50000 internal come from? Would it work for any model?

Reply:
Good point. The OPT models employ a vocabulary whose size exceeds 50K, and I thought the inference time might depend on the inputs (hence the uniform sampling). After testing, I don't think that's the case, though, so let me change it back to fixed prompts.